code.js

```diff
@@ -1,3 +1,79 @@
 function depthFirstSearch(graph, startNode, targetNode) {
+    
+    let isThere = false;
+    let stack = [];
+    let visited = [];
+    stack.push(startNode);
+    
+    while (visited.length < graph.length) {
+        console.log("new loop");
+        visited.push(stack[stack.length - 1]);
+        if (stack[stack.length - 1] == targetNode) {
+            return visited;
+        }
+        
+        let tmp = stack[stack.length - 1];
+        stack.pop();
+        for (let i = 0; i < graph[tmp].length; i++) { //look at adjacent nodes
+            if (visited[i] == "") { //if not visited then add to the stack
```

Comment:
This was just me testing a way to make checking visited more efficient. I did not mean to have it request review. 

I have tried changing it to check if visited at the adjacent node is null or not but I cant get it to work. It gets part way and then throws an error because it is not added to the stack. I am unsure how to make it more efficient.

Reply:
There's a function that checks whether a list includes an item. You could also use a hash map or similar.